### PR TITLE
Add Niagara College scraper

### PR DESCRIPTION
**Pull request by jodizzle**
https://ess.niagaracollege.ca/CourseOutline/Default.aspx

Pretty straightforward.  I'm not sure if a docstring is super necessary here, but I decided to write one anyway.

It seems like every syllabi page is loaded after both a 301 and a 302, so as of right now downloads during the actual run will probably take a long time, as per #69.

### CODE REVIEW

File: osp_scraper/spiders/niagaracollege.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    selecting an option for the previous field.  To simulate this,
+    `dont_click=True` is used.
+    """
+
+    name = "niagaracollege"
+
+    start_urls = ["https://ess.niagaracollege.ca/CourseOutline/Default.aspx"]
+
+    def parse(self, response):
+        subjects = response.css("#ctl00_ContentPlaceholderMain_ListBoxLookupSubject")\
```
Comment (HarrisonGregg) [inline osp_scraper/spiders/niagaracollege.py]: Is there a reason you didn't add a line break before the first `.css`?

File: osp_scraper/spiders/niagaracollege.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    def parse_for_catalogue_numbers(self, response):
+        numbers = response.css("#ctl00_ContentPlaceholderMain_ListBoxLookupCatalogueNumber")\
+                          .css("option::attr(value)")[1:].extract()
+        for number in numbers:
+            yield scrapy.FormRequest.from_response(
+                response,
+                method="POST",
+                formdata={
+                    '__EVENTTARGET': 'ctl00$ContentPlaceholderMain$ListBoxLookupCatalogueNumber',
+                    'ctl00$ContentPlaceholderMain$ListBoxLookupCatalogueNumber': number
```
Comment (HarrisonGregg) [inline osp_scraper/spiders/niagaracollege.py]: Could add a line break after the `:` if you want.

File: osp_scraper/spiders/niagaracollege.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+                callback=self.parse_for_terms
+            )
+
+    def parse_for_terms(self, response):
+        terms = response.css("#ctl00_ContentPlaceholderMain_ListBoxLookupAcademicTerm")\
+                        .css("option")
+        for term in terms:
+            term_code = term.css('::attr(value)').extract_first()
+            term_text = term.css('::text').extract_first()
+            anchor = '{subject} {number}'.format(**response.meta) + " " + term_text
```
Comment (HarrisonGregg) [inline osp_scraper/spiders/niagaracollege.py]: Seems a little awkward to write it like this.  Alternatives could be 
```python
            anchor = '{subject} {number} {0}'.format(term_text, **response.meta)
```
or
```python
            anchor = " ".join([
                response.meta['subject'], 
                response.meta['number'],
                term_text
            ])
```

File: osp_scraper/spiders/niagaracollege.py (hunk 1 above)
Comment (jodizzle) [inline osp_scraper/spiders/niagaracollege.py]: Not in particular.  I figure the string is gonna be pretty long and pushing the line-length boundaries anyway, so I just put it on the first line.  But I guess I could re-arrange.

File: osp_scraper/spiders/niagaracollege.py (hunk 2 above)
Comment (jodizzle) [inline osp_scraper/spiders/niagaracollege.py]: Agreed.

File: osp_scraper/spiders/niagaracollege.py (hunk 3 above)
Comment (jodizzle) [inline osp_scraper/spiders/niagaracollege.py]: I think the second one looks nice.

**Pull request merged**